### PR TITLE
refactor: upgrade django to 3.0.8

### LINKED ISSUE #565: Upgrade Django to latest version
## Description
As a developer,
I need upgrade Django.
so that portal is up to date with the latest dependency.

## Acceptance Criteria
Django is updated to latest stable version.

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
4 hours

### PR DESCRIPTION
**Pull request by satya7289**
### Description
Upgrade django to 3.0.8. 
major changes for upgrading to 3.0.8 are as follows
- update `from django.core.urlresolvers import reverse` -> `from django.urls import reverse`
- add `on_delete=models.CASCADE` to foreignkey and onetoonekey relationship for the model.
- update requirement files.
- remove old migration to latest one(because some of the migrations are created with python2.7 which conflict with python3 or django 2.0)

Fixes #565

### Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)

### How Has This Been Tested?
pass all test cases. running fine locally.
Screenshot from 2020-07-14 14-27-00

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: requirements/prod.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-Django==1.11.29
+Django==2.0
 Pillow==4.3.0
 django-allauth==0.33.0
-django-braces==1.11.0
-django-cities-light==3.3.0
+django-braces==1.14.0
+django-cities-light==3.6.0
```
Comment (SanketDG) [inline requirements/prod.txt:5]: We also should upgrade as many dependencies as we can, we don't want to ship packages that have vulnerabilities

Also https://pyup.io/safety/ is a great tool to check which requirements have safety issues

**Review by SanketDG — CHANGES_REQUESTED** [review]
I don't remember deciding with you or @sakshi1499 that we are going to stick to using version 2. Django 2.0 was almost released two years ago and has already reached support end https://www.djangoproject.com/download/#supported-versions

We can go with 2.2 LTS, but given Anita-B is a fast moving organization it makes no sense to use an older LTS version.

I would like to know what problems you faced with upgrading to 3.0, and we can all try to fix them.

File: requirements/prod.txt (hunk 1 above)
Comment (satya7289) [reply requirements/prod.txt:5]: Ok

**Review by theyashshahs — CHANGES_REQUESTED** [review]
@satya7289 remove the migrations from the folder

**Comment by satya7289** [comment]
> @satya7289 remove the migrations from the folder

I have to modify the migrations file for upgrading django to latest version(as some of the migrations where created using python2.7 which is not supported by django latest version). [see](https://anitab-org.zulipchat.com/#narrow/stream/222540-portal/topic/Update-Django)

**Review by theyashshahs — COMMENTED** [review]
@satya7289 can you provide a GIF or a screenshot of the working :)

**Comment by SanketDG** [comment]
@satya7289 Did you make the changes as we discussed for Django 3.0?

File: .travis.yml [hunk 2]
```diff
@@ -44,4 +47,4 @@ notifications:
     on_failure: always
     on_start: false
 after_success:
-  coveralls --rcfile=.coveragerc
+  coveralls --rcfile=.coveragerc
```
Comment (SanketDG) [inline .travis.yml:50]: you removed the newline

**Comment by SanketDG** [comment]
Great work @satya7289 !

Why did you add html file changes though? They cannot be part of this PR.

**Comment by satya7289** [comment]
> Great work @satya7289 !
> 
> Why did you add html file changes though? They cannot be part of this PR.

{% load staticfiles %} has been changed to {% load static %} for django 3.0.8

File: .travis.yml (hunk 2 above)
Comment (satya7289) [reply .travis.yml:50]: Does it metter here, as it is not .py file!!!

File: setup.cfg [hunk 3]
```diff
@@ -1,4 +1,4 @@
 [flake8]
-ignore = F403
+ignore = F403, F405
```
Comment (SanketDG) [inline setup.cfg:2]: What's the need for adding this?

File: systers_portal/templates/common/contact.html [hunk 4]
```diff
@@ -19,4 +20,4 @@ <h1 class="contact-head">Contact Systers</h1>
     </div>
     <div class="mt40"></div>
   </div>
-{% endblock %}
+{% endblock %}
```
Comment (SanketDG) [inline systers_portal/templates/common/contact.html]: add the newlines back

**Review by SanketDG — CHANGES_REQUESTED** [review]
A few nitpicks.

Also the migrations were generated with Django 2, can you re-generate them with Django 3?

File: setup.cfg (hunk 3 above)
Comment (satya7289) [reply setup.cfg:2]: Due to import * is used some files. Exactly not remember which files.

Comment (SanketDG) [reply setup.cfg:2]: I don't see any star imports though

**Review by SanketDG — APPROVED** [review]
This looks really good!

@satya7289 Excellent work, and patience!

**Review by sakshi1499 — APPROVED** [review]

**Comment by SanketDG** [comment]
:tada: :tada: 

This is great @satya7289 

I have opened https://github.com/anitab-org/portal/issues/622 (will update it soon) for some warnings during the test run

**Pull request merged**